### DISCUSSION TITLE
Automatic redirect from a path ending with trailing slash to the same path without trailing slash

### DISCUSSION THREAD

**Discussion by cristobal** [fastify/fastify#4782] [General] [answered]
As far as i can see i can't see that there is any route option that allows for automatic redirect with `302` from routes that have trailing slash to the same route without a trailing slash e.g. redirec from:
* `/some/path/` 
to:
* `/some/path`

I can fix this manually in my route handlers or in a route hook, however i was wondering if this is something that has been discussed before and if there was some reason that this has not been implemented. I remember having read about some redirection attacks...

The reason i ask is that we only want to have one route registered in our web application, however we don't always control outside links to our own web application, and we ofte have links that do this.

**Comment by jsumners** [comment]
> i was wondering if this is something that has been discussed before and if there was some reason that this has not been implemented.

No, it hasn't been discussed to my recollection. We really only have support for mapping `/foo/` to the `/foo` handler.

Personally, I think the desired redirect is better handled at the reverse proxy.

**Reply by cristobal** [reply]
Yes that’s a valid point still under some circumstances one are not always behind an reverse proxy.

Also adding logic from the web application into a reverse proxy may not always be desirable. If we only want todo so for some routes but not all of them, then application logic starts leaking into the web proxy.

**Reply by cristobal** [reply]
There are also some cases in our web application, where we want to resolve manually due to some legacy url support and some other paths were we want to have it automatically.

So i thought since this is an feature that does not exists today, that it could be added as part of the route optionns.  And i could make an PR for it. That’s why i was asking.

However if there’s no desire to add this as an an route option, i will just create an package for it instead 👋

**Reply by metcoder95** [reply, marked as answer]
Agree with @jsumners that most of this logic should belong to an abstraction layer on top of fastify, like a reverse proxy or API gateway, or any other strategy. Though, I can also understand that some applications might have the desire to have it within the app.

I honestly don't think that it should land on the core, as it sounds like an issue that can cause some maintenance overhead (to not mention the number of issues that can come with it). Having it as a `plugin` sounds like a good way to go 👍 
You can create the plugin and later open a PR for adding it to the community section.

**Reply by cristobal** [reply]
@metcoder95 yes we use a CDN provider, were the recommendation is actually better to let the application handle the resolve with a proper redirect code and caching header.

**Reply by cristobal** [reply]
Thanks for the input @jsumners @metcoder95, marked this as resolved.
